src/track/track.cpp

```diff
@@ -1403,47 +1439,6 @@ ExportTrackMetadataResult Track::exportMetadata(
         return ExportTrackMetadataResult::Skipped;
     }
 
-    if (pConfig->getValue<bool>(kConfigKeySeratoMetadataExport)) {
-        const auto streamInfo = m_record.getStreamInfoFromSource();
-        VERIFY_OR_DEBUG_ASSERT(streamInfo &&
-                streamInfo->getSignalInfo().isValid() &&
-                streamInfo->getDuration() > mixxx::Duration::empty()) {
-            kLogger.warning() << "Cannot write Serato metadata because signal "
-                                 "info and/or duration is not available:"
-                              << getLocation();
-            return ExportTrackMetadataResult::Skipped;
```

Comment:
The early return on error gets lost here unfortunately. I don't know the surrounding code well enough to judge whether thats an issue.

Reply:
This is intended. Exporting Serato is only a minor feature, failing should not fail the main metadata export. 
We are here in a DEBUG_ASSERT branch, that never happens anyway. A last resort protection against exporting an inconsistent Serato Tag.

Reply:
Mhmm. I think it would still make sense to add this information as the return value. Why not keep the `return ExportTrackMetadataResult::Skipped` and return `ExportTrackMetadataResult` instead of `void`. The caller could then check or otherwise handle the issue. That way `exportSeratoMetadata` is not entirely the black box is currently is.

Reply:
I have removed it, because it was unused code after the changes in this PR. Currently we print a warning for this case (that must never happen). What else do you whish to do in the caller?

Reply:
Nothing really. With the current decision to not abort the export when exporting serato metadata failed the return value would be unused.  But just looking at the refactoring from an API perspective, having a function that can fail return some indication of failure instead of doing nothing makes more sense to me, even if the caller chooses to ignore the failure. *That decision is up the caller and should not be made by the callee for all callers,* hence it would be appropriate to return a result of the operation.

Reply:
Done

Reply:
IMO returning `ExportTrackMetadataResult` would've been more appropriate, but I will stop bikeshedding now.